tests/streams/test_memory.py

```diff
@@ -169,9 +169,15 @@ async def test_cancel_receive() -> None:
         await wait_all_tasks_blocked()
         tg.cancel_scope.cancel()
 
+    assert receive._state.is_closed
+    assert send._state.is_closed
+
     with pytest.raises(WouldBlock):
         send.send_nowait("hello")
 
+    with pytest.raises(EndOfStream):
+        await receive.receive()
```

Comment:
This is just plain **wrong**. The send stream was never closed, so why would `receive.receive()` raise `EndOfStream`?